src/App.js

```diff
@@ -3,28 +3,21 @@ import Tweet from './CabinList';
 import cabinData from './data/cabin';
 
 function App(){
-  const cabinNameList = cabinData.map((data, index) => (
-    <li key={index}>{data.cabin} </li>
-  ))
-  return (
-      <ul>
-        {cabinData.map((data, index) => (
-          <li key={index}>{data.cabin} </li>
-        ))}
-      </ul>
-      
-  );
-  // const cabinNameList = cabinData.map((data) => {
-  //   return <Tweet cabin={data.cabin} />
-  // });
-  // return(
-  //   <div className='app'>
-  //     <ul>
-  //       <li>{cabinNameList} </li>
-  //     </ul>
-  //     {/* {cabinNameList} */}
-  //   </div>
-  // )
+
+const cabinNameList1 = cabinData.map(data => data.cabin)
+const cabinNames = [...new Set(cabinNameList1)] 
+const cabin = cabinNames.map((data, index)=> (
+  <li key={index}>
+  <button>{data} </button>
+  </li>
+  // console.log();
+  // <li key={index}>{data}  </li>
+))
+return (
+  <ul>
+    {cabin}
+  </ul>
```

Comment:
this is optional, what you've got is totally valid, but you can argue that because your list item is such a small piece of code, that mapping through the cabinNames inline here is more 'readable' as the HTML is all visible together e.g.

```
  return (
    <ul>
      { cabinNames.map((data, index)=> (
        <li key={index}>
          <button>{data}</button>
        </li>
        }
    </ul>
  )
```

what you've got at the moment (e.g. creating a separate const) might be preferable when there's lots of complicated logic required in order to create the cabinNames list, but, you could then also split that into a separate component.